src/main/kotlin/uk/gov/gdx/datashare/repositories/AcquirerSubscriptionRepository.kt

```diff
@@ -10,28 +10,42 @@ import java.util.*
 @Repository
 @JaversSpringDataAuditable
 interface AcquirerSubscriptionRepository : CrudRepository<AcquirerSubscription, UUID> {
-  fun findAllByOauthClientId(oauthClientId: String): List<AcquirerSubscription>
+  @Override
+  override fun findAll() = findAllByWhenDeletedIsNull()
+  fun findAllByWhenDeletedIsNull(): List<AcquirerSubscription>
 
-  fun findAllByEventType(eventType: EventType): List<AcquirerSubscription>
+  fun findAllByOauthClientId(oauthClientId: String) = findAllByOauthClientIdAndWhenDeletedIsNull(oauthClientId)
+  fun findAllByOauthClientIdAndWhenDeletedIsNull(oauthClientId: String): List<AcquirerSubscription>
+
+  fun findAllByEventType(eventType: EventType) = findAllByEventTypeAndWhenDeletedIsNull(eventType)
+  fun findAllByEventTypeAndWhenDeletedIsNull(eventType: EventType): List<AcquirerSubscription>
+
+  fun findAllByQueueName(queueName: String) = findAllByQueueNameAndWhenDeletedIsNull(queueName)
+  fun findAllByQueueNameAndWhenDeletedIsNull(queueName: String): List<AcquirerSubscription>
 
   fun findByAcquirerSubscriptionIdAndQueueNameIsNotNull(acquirerSubscriptionId: UUID): AcquirerSubscription?
 
   fun findAllByOauthClientIdAndEventTypeIsIn(
     oauthClientId: String,
     eventTypes: List<EventType>,
+  ) = findAllByOauthClientIdAndWhenDeletedIsNullAndEventTypeIsIn(oauthClientId, eventTypes)
+
+  fun findAllByOauthClientIdAndWhenDeletedIsNullAndEventTypeIsIn(
+    oauthClientId: String,
+    eventTypes: List<EventType>,
   ): List<AcquirerSubscription>
 
   @Query(
     "SELECT asub.* FROM acquirer_subscription asub " +
       "JOIN acquirer_event ae ON asub.id = ae.acquirer_subscription_id " +
-      "WHERE ae.id = :id ",
+      "WHERE ae.id = :id AND asub.when_deleted IS NULL",
   )
   fun findByEventId(id: UUID): AcquirerSubscription?
 
   @Query(
     "SELECT asub.* FROM acquirer_subscription asub " +
       "JOIN acquirer a ON asub.acquirer_id = a.id " +
-      "AND a.id = :id",
+      "AND a.id = :id AND asub.when_deleted IS NULL",
```

Comment:
Just spotted should this first `AND` be a `WHERE`?

Reply:
Not necessarily. There's nothing wrong with this being an additional condition on the join: https://stackoverflow.com/questions/1018952/condition-within-join-or-where. I'd say your addition definitely wants to be a `WHERE` and it might make sense to make both conditions part of the WHERE clause